src/methods/checkLang.ts

```diff
@@ -1,28 +1,23 @@
 "use strict"
 
-export declare namespace checkLang {
+declare namespace checkLang {
   type langCode = "ja" | "en"
 }
 
-export const checkLang = (): checkLang.langCode => {
-  const mylang = document.getElementById("mylang")
+const getCheckLangElement = () => document.getElementById("mylang")
 
-  if (mylang) {
-    if (mylang.className.includes("mylang-ja")) {
-      return "ja"
-    } else if (mylang.className.includes("mylang-en")) {
-      return "en"
-    } else if (mylang.innerText.includes("English")) {
-      // mylangのクラス名がサービス側で変更されたケースがあったため、フォールバック
-      return "ja"
-    } else if (mylang.innerText.includes("日本語")) {
-      // mylangのクラス名がサービス側で変更されたケースがあったため、フォールバック
-      return "en"
-    } else {
-      // 日本語をデフォルトに
-      return "ja"
-    }
-  }
-
-  return "ja"
+const checkLangByDOMElement = (
+  mylang: HTMLElement
+): checkLang.langCode | undefined => {
+  if (mylang.classList.contains("mylang-ja")) return "ja"
+  if (mylang.classList.contains("mylang-en")) return "en"
+  if (mylang.innerText.includes("日本語")) return "ja"
+  if (mylang.innerText.includes("English")) return "en"
 }
+
+const checkLang = (
+  checkLangElement: HTMLElement | null = getCheckLangElement()
```

Comment:
What is your intention on recieving this as a props?

Reply:
To be honest, this is the middle ground for this function not to take a block. But this could expose this function to being assigned an unexpected property.
Should I rewrite it to:
```ts
const checkLang = (): checkLang.langCode => {
  const checkLangElement = document.getElementById("mylang")
  return (checkLangElement && checkLangByDOMElement(checkLangElement)) ?? "ja"
}
```

Reply:
It feels like a way too much to change the behavior (exposed to outside the module) just for reducing one code block. The snippet you suggested seems moderate.